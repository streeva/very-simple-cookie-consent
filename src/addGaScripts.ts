export default (tag: string) => {
  addHeadScript(tag)
  addBodyNoscript(tag)
}

const addHeadScript = (tag: string) => {
  if (document.querySelector('#vs-cc-script')) return

  const script = document.createElement('script')
  script.id = 'vs-cc-script'
  script.innerHTML = `
  (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
  new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
  j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
  'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
  })(window,document,'script','dataLayer','${tag}');
  `

  document.head.appendChild(script)
}

const addBodyNoscript = (tag: string) => {
  if (document.querySelector('#vs-cc-noscript')) return

  const noscript = document.createElement('noscript')
  noscript.id = 'vs-cc-noscript'
  const iframe = document.createElement('iframe')
  iframe.src = `https://www.googletagmanager.com/ns.html?id=${tag}`
  iframe.width = '0'
  iframe.height = '0'
  iframe.setAttribute('style', 'display:none;visibility:hidden')
  noscript.appendChild(iframe)
  document.body.insertBefore(noscript, document.body.firstChild)
}
