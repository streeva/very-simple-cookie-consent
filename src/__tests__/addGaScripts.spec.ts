import { expect, test } from 'vitest'
import addGaScripts from '../addGaScripts'

test('addGaScripts', () => {
  const tag = 'GA-123'

  addGaScripts(tag)

  expect(document.body.toString()).toMatchSnapshot()
  expect(document.head.toString()).toMatchSnapshot()
})
