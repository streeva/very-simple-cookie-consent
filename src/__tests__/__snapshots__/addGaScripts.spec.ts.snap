// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`addGaScripts 1`] = `"<body><noscript id="vs-cc-noscript"><iframe src="https://www.googletagmanager.com/ns.html?id=GA-123" width="0" height="0" style="display:none;visibility:hidden"></iframe></noscript></body>"`;

exports[`addGaScripts 2`] = `
"<head><script async="" src="https://www.googletagmanager.com/gtm.js?id=GA-123"></script><script id="vs-cc-script">
  (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
  new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
  j=d.createElement(s),dl=l!='dataLayer'?'&amp;l='+l:'';j.async=true;j.src=
  'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
  })(window,document,'script','dataLayer','GA-123');
  </script></head>"
`;
